google/gax/__init__.py

```diff
@@ -588,10 +590,10 @@ def result(self, timeout=None):
 
     def exception(self, timeout=None):
         """Similar to result(), except returns the exception if any."""
-        if self._poll(timeout).HasField('error'):
-            return self._operation.error
+        if not self._poll(timeout).HasField('error'):
```

Comment:
I could go either way on the positive/negative `if` nitpicks. I agree that positive `if` blocks are a bit easier to follow, but I _also_ like the pattern where the "normal" / expected return value is at the end, and the if-block is the "aside case". I am _guessing_ that is what @eoogbe was going for.

If this were my code, I would have probably done it her way, but with a bit more narrative:

```python
def exception(self, timeout=None):
    """Return the exception, if any, or None otherwise."""

    # Sanity check: If there is no current exception, then just return None.
    if not self._poll(timeout).HasField('error'):
        return None

    # Okay, there was an exception as expected; return it.
    return self._operation.error
```